### PR TITLE
[microNPU] Introduce a pass to remove redundant identity operations

### PR DESCRIPTION
**Pull request by lhutton1**
Introduces a pass that aims to remove identity operations, introduced during legalization, that are immediately followed by an NPU compute operation e.g. Convolution.

cc @manupa-arm @ekalda @mbaret @NicolaLancellotti @jacobbohlin @dchauhan-arm

### CODE REVIEW

**Review by ekalda — COMMENTED** [review]
Thanks for the patch, looks really good so far! :) Maybe one testcase to add (to the already very comprehensive set) would be to check that there are no reshape or strided slice directly followed by concatenate.

File: tests/python/contrib/test_ethosu/test_identity_optimizer.py [hunk 1]
```diff
@@ -0,0 +1,245 @@
...
+from .test_codegen import _compare_tvm_with_tflite
+
+
+def _optimize(expr, optimize=True):
+    """Create IRModule and run layout optimizer pass."""
+    mod = tvm.IRModule.from_expr(expr)
+    mod = relay.transform.InferType()(mod)
+    if optimize:
+        mod = IdentityOptimizer()(mod)
+    print(mod)
```
[the remark this answers is not in the document]
Comment (lhutton1) [reply tests/python/contrib/test_ethosu/test_identity_optimizer.py]: Removed, thanks!

**Comment by lhutton1** [comment]
Thanks for the review @ekalda, I added a test case and removed the print among a couple more things I spotted :)

**Review by ekalda — APPROVED** [review]
Thanks @lhutton1, looks good to me! :)

**Comment by lhutton1** [comment]
So it turns out supporting the removal of identity operations with multiple outputs was not as complicated as I first thought, and it was technically already supported in the original patch. Therefore, `CountIdentityOutputs` and the restriction in `RemoveRedundantIdentities` can be removed. Added tests for the multiple output case to support this. Apologies for the confusion @ekalda, could you take another look when you have time :)

**Review by ekalda — APPROVED** [review]
Thanks @lhutton1 , looks all good to me! :)

**Review by manupa-arm — APPROVED** [review]
LGTM!

**Comment by manupa-arm** [comment]
Thanks @ekalda @lhutton1 !

**Pull request merged**